command/role.go

```diff
@@ -31,16 +30,43 @@ func role(ctx *Context, args []string) {
 		return
 	}
 
-	number, err := strconv.Atoi(args[1])
-	if err != nil {
-		ctx.ReportError("Invalid number", err)
-		return
-	} else if number < 0 || number >= len(ctx.Env.RoleColors) {
-		ctx.Reply("Invalid number")
-		return
+	type colorRole struct {
+		ID   string `json:"id"`
+		Name string `json:"color"`
+	}
+	var colorRoles []colorRole
+	for i, _ := range ctx.Env.RoleColors {
+		color, _ := ctx.Session.State.Role(ctx.Message.GuildID, ctx.Env.RoleColors[i])
+		name := fmt.Sprintf("%s", (color.Name))
+		r := colorRole{
+			ID:   color.ID,
+			Name: name,
+		}
+		colorRoles = append(colorRoles, r)
```

Comment:
I think this initialization should be done during startup, not everytime the command is run.

Reply:
got it

Reply:
NVM, I don't know how I would implement this

Reply:
I don't think you get to just loop over RoleColors, I think we just store IDs in the environment. So for now, the environment variable either needs to be duplicated(to include colors and ids), or changed, or just left. 
@mlvzk thoughts?

Reply:
You could intialize the color role structures in main.go's session ready handler - `discord.AddHandler(func(s *discordgo.Session, r *discordgo.Ready) {`, and put it in the Env struct(will require changing the type of RoleColors)

Reply:
okay